moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryMap.js

```diff
@@ -1,107 +1,80 @@
-import React, { useRef, useCallback, useState } from "react";
-import Map, { NavigationControl } from "react-map-gl";
-import { Box, makeStyles } from "@material-ui/core";
-import "mapbox-gl/dist/mapbox-gl.css";
-
+import React, { useCallback, useState, useMemo } from "react";
+import MapGL from "react-map-gl";
+import { Box } from "@material-ui/core";
+import ProjectSummaryMapFallback from "./ProjectSummaryMapFallback";
+import BaseMapSourceAndLayers from "../ProjectComponents/BaseMapSourceAndLayers";
+import BasemapSpeedDial from "../ProjectComponents/BasemapSpeedDial";
+import ProjectSummaryMapSourcesAndLayers from "./ProjectSummaryMapSourcesAndLayers";
 import {
-  createSummaryMapLayers,
-  getSummaryMapInteractiveIds,
-  MAPBOX_TOKEN,
-  mapStyles,
-  renderTooltip,
-  countFeatures,
-  useHoverLayer,
-  useFeatureCollectionToFitBounds,
   basemaps,
-  mapConfig,
-} from "../../../../utils/mapHelpers";
+  mapParameters,
+  initialViewState,
+} from "../ProjectComponents/mapSettings";
+import { makeFeatureFromProjectGeographyRecord } from "../ProjectComponents/utils/makeFeatureCollections";
+import { useZoomToExistingComponents } from "../ProjectComponents/utils/map";
+import "mapbox-gl/dist/mapbox-gl.css";
+
+/**
+ * Use a callback ref to get the map instance and store it in state so we can watch it with useEffect
```

Comment:
This solves a bug that I was facing with nothing in this component triggering the `useEffect` in `useZoomToExistingComponents`. In a nutshell, this callback gives us both the normal React ref behavior and also the ref value in state so that we can watch it for updates.

It revolves around how React refs hold the same value throughout a component's lifecycle but they do not trigger a re-render when that value is updated. Here is [a thread](https://twitter.com/teh_builder/status/1093497029693763584) that I always find my way back to on this. This type of stuff is a head-scratcher for me every time.

Also, if you are wondering why it works in the component map, it is because there is an async call for `data` there so that is what triggers the zoom to features. `data` has already been fetched here and is passed as a prop so it doesn't change/trigger `useEffect`.

Reply:
very cool! so glad to see this resolved!